transaction-view/src/address_table_lookup_meta.rs

```diff
@@ -0,0 +1,167 @@
+use {
+    crate::{
+        bytes::{
+            advance_offset_for_array, advance_offset_for_type, check_remaining,
+            optimized_read_compressed_u16, read_byte,
+        },
+        result::Result,
+    },
+    solana_sdk::{packet::PACKET_DATA_SIZE, pubkey::Pubkey},
+};
+
+/// Contains meta-data about the address table lookups in a transaction packet.
+pub struct AddressTableLookupMeta {
+    /// The number of address table lookups in the transaction.
+    pub(crate) num_address_table_lookup: u16,
+    /// The offset to the first address table lookup in the transaction.
+    pub(crate) offset: u16,
+}
+
+impl AddressTableLookupMeta {
+    /// Get the number of address table lookups (ATL) and offset to the first.
+    /// The offset will be updated to point to the first byte after the last
+    /// ATL.
+    /// This function will parse each ATL to ensure the data is well-formed,
+    /// but will not cache data related to these ATLs.
+    pub fn try_new(bytes: &[u8], offset: &mut usize) -> Result<Self> {
+        // Read the number of ATLs at the current offset.
+        // Each ATL needs at least 34 bytes, so do a sanity check here to
+        // ensure we have enough bytes to read the number of ATLs.
+        const MIN_SIZED_ATL: usize = core::mem::size_of::<Pubkey>() + 1 + 1;
```

Comment:
```suggestion
        const MIN_SIZED_ATL: usize = core::mem::size_of::<Pubkey>() + 2;
```

Reply:
c0d298dc65c7fb06d5d58caca17fc32467da77dc

I added comments to each number to clarify what it is representing.